packages/components/src/button/use-deprecated-button-props.js

```diff
@@ -0,0 +1,195 @@
+/**
+ * Internal dependencies
+ */
+import Dashicon from '../dashicon';
+
+const getVariant = ( {
+	variant,
+	isPrimary,
+	isTertiary,
+	isSecondary,
+	isLink,
+	isDefault,
+} ) => {
+	if ( typeof variant !== 'undefined' ) {
+		return variant;
+	}
+
+	if ( isPrimary ) {
+		deprecated( 'Button isPrimary', {
+			alternative: 'variant="primary"',
+		} );
+		return 'primary';
+	}
+
+	if ( isTertiary ) {
+		deprecated( 'Button isTertiary', {
+			alternative: 'variant="tertiary"',
+		} );
+		return 'tertiary';
+	}
+
+	if ( isSecondary ) {
+		deprecated( 'Button isSecondary', {
+			alternative: 'variant="secondary"',
+		} );
+		return 'secondary';
+	}
+
+	if ( isDefault ) {
+		deprecated( 'Button isDefault', { alternative: 'variant="plain"' } );
+		return 'plain';
+	}
+
+	if ( isLink ) {
+		deprecated( 'Button isLink', { alternative: 'variant="link"' } );
+		return 'link';
+	}
+
+	return 'plain';
+};
+
+function getTooltip( {
+	tooltip,
+	describedBy,
+	ariaLabel,
+	label,
+	tooltipPosition,
+	shortcut,
+} ) {
+	if ( typeof tooltip !== 'undefined' ) {
+		return tooltip;
+	}
+
+	tooltip = {};
+
+	if ( tooltipPosition ) {
+		deprecated( 'Button tooltipPosition', {
+			alternative: 'tooltip={{ position: tooltipPosition }}',
+		} );
+		tooltip.position = tooltipPosition;
+	}
+
+	if ( shortcut ) {
+		deprecated( 'Button shortcut', {
+			alternative: 'tooltip={{ shortcut }}',
+		} );
+		tooltip.shortcut = shortcut;
+	}
+
+	if ( Object.keys( tooltip ).length !== 0 ) {
+		// if there is indeed deprecated tooltip props being passed in
+		deprecated( 'Button implicit tooltip content', {
+			alternative: 'tooltip={{ content }}',
+		} );
+		tooltip.content = describedBy ? describedBy : ariaLabel || label;
+	}
+
+	return tooltip;
+}
+
+function getPre( { pre, text } ) {
+	if ( typeof text !== 'undefined' ) {
+		deprecated( 'Button text', { alternative: 'pre={ text }' } );
+		return text;
+	}
+	return pre;
+}
+
+function getAriaLabel( { ariaLabel, label } ) {
+	if ( typeof label !== 'undefined' ) {
+		deprecated( 'Button label', { alternative: 'aria-label={ label }' } );
+		return label;
+	}
+	return ariaLabel;
+}
+
+function getSize( { isSmall, size } ) {
+	if ( typeof isSmall !== 'undefined' ) {
+		deprecated( 'Button isSmall', { alternative: 'Button size' } );
+		return isSmall ? 'small' : undefined;
+	}
+	return size;
+}
+
+function getIsActive( { isBusy, isActive } ) {
+	if ( typeof isBusy !== 'undefined' ) {
+		deprecated( 'Button isBusy', { alternative: 'Button isActive' } );
+		return isBusy;
+	}
+	return isActive;
+}
+
+function getIsLoading( { isBusy, isLoading } ) {
+	if ( typeof isBusy !== 'undefined' ) {
+		deprecated( 'Button isBusy', { alternative: 'Button isLoading' } );
+		return isBusy;
+	}
+	return isLoading;
+}
+
+function getIcon( { icon } ) {
+	if ( typeof icon === 'string' ) {
+		deprecated( 'Button icon as string', {
+			alternative: 'Button icon as ReactElement',
+			hint: 'Use the icons from @wordpress/icons',
+		} );
```

Comment:
I know for instance that this is not something we can officially deprecate yet. Blocks can use string icons and they can't rely on svg or packages in a lot of situations.

Reply:
Okay! In that case we can just remove the deprecation warning, right?

Reply:
yes :)